### ISSUE TITLE
HTTPResponseCode enum accidentally cast to int value as param of intToHTTPResponseCode

### ISSUE THREAD

**Issue by jbfitb** [alexa/avs-device-sdk#1519]
## Briefly summarize your issue:
[It looks like `finalResponseCode` of type `HTTPResponseCode` is accidentally being cast as an `int` in the function `intToHTTPResponseCode(...)`:](https://github.com/alexa/avs-device-sdk/blob/master/AVSCommon/Utils/src/LibcurlUtils/LibCurlHttpContentFetcher.cpp#L370)

```
int finalResponseCodeId = 0;
auto curlReturnValue = 
    curl_easy_getinfo(m_curlWrapper.getCurlHandle(), CURLINFO_RESPONSE_CODE, &finalResponseCodeId);
finalResponseCode = intToHTTPResponseCode(finalResponseCode);
```

I believe the intention of the code was to get the current response code and then convert that into the `HTTPResponseCode` enum, and that this line:

```
finalResponseCode = intToHTTPResponseCode(finalResponseCode);
```

should be

```
finalResponseCode = intToHTTPResponseCode(finalResponseCodeId);
```

To support this theory, It looks like `finalResponseCodeId` is never referenced after `curl_easy_getinfo` writes to it.

### Ramifications

Without this change, the content fetcher will interpret the response code as 0, which will return `HTTPResponseCode::HTTP_RESPONSE_CODE_UNDEFINED`.

**Comment by scotthea-amazon** [comment]
Hello @jbfitb,

This does appear to be a bug.  Thank you for bringing this to our attention! 

Best regards,
-SWH

**Comment by scotthea-amazon** [comment]
Hello @jbfitb,

This has been fixed in the 1.17 release.  Thank you again for you help!

Best regards,
-SWH
